hpx/parallel/algorithms/copy.hpp

```diff
@@ -324,6 +331,21 @@ namespace hpx { namespace parallel { HPX_INLINE_NAMESPACE(v1)
     // copy_if
     namespace detail
     {
+        // sequential copy_if with projection function
+        template <typename InIter, typename OutIter, typename Pred, typename Proj>
+        inline OutIter
+        sequential_copy_if(InIter first, InIter last, OutIter dest,
+            Pred && pred, Proj && proj)
+        {
+            while (first != last)
+            {
+                if (hpx::util::invoke(pred, hpx::util::invoke(proj, *first)))
+                    *dest++ = *first;
```

Comment:
I find it strange that the projection is not applied when assigning the iterator, only for the predicate. I looked it up in the extensions paper and the implementation seems to be consistent with the spec. Is this a defect in the spec?